pkg/sql/tests/rsg_test.go

```diff
@@ -333,11 +369,80 @@ func TestRandomSyntaxSchemaChangeColumn(t *testing.T) {
 	})
 }
 
+var ignoredErrorPatterns = []string{
+	"unimplemented",
+	"unsupported binary operator",
+	"unsupported comparison operator",
+	"memory budget exceeded",
+	"generator functions are not allowed in",
+	"txn already encountered an error; cannot be used anymore",
+	// Numeric conditions
+	"exponent out of range",
+	"result out of range",
+	"argument out of range",
+	"integer out of range",
+	"invalid operation",
+	"invalid mask",
+	"cannot take square root of a negative number",
+	"out of int64 range",
+	"underflow, subnormal",
+	"overflow",
+	// Type checking
+	"value type .* doesn't match type .* of column",
+	"incompatible value type",
+	"incompatible COALESCE expressions",
+	"error type checking constant value",
+	"ambiguous binary operator",
+	"ambiguous call",
+	"cannot be matched",
+	"unknown signature",
+	"cannot determine type of empty array",
+	"conflicting ColumnTypes",
+	// Data dependencies
+	"violates not-null constraint",
+	"violates unique constraint",
+	// Context-specific string formats
+	"invalid regexp flag",
+	"unrecognized privilege",
+	"invalid escape string",
+	"error parsing regexp",
+	"could not parse .* as type bytes",
+	"UUID must be exactly 16 bytes long",
+	"unsupported timespan",
+	"does not exist",
+	"unterminated string",
+	"incorrect UUID length",
+	"the input string must not be empty",
+	// JSON builtins
+	"mismatched array dimensions",
+	"cannot get array length of a non-array",
+	"cannot be called on a non-array",
+	"cannot call json_object_keys on an array",
+	// Builtins that have funky preconditions
+	"cannot delete from scalar",
+	"lastval is not yet defined",
+	"negative substring length",
+	"non-positive substring length",
+	"bit strings of different sizes",
+	"inet addresses with different sizes",
+	"values of different sizes",
+	"must have even number of elements",
+	"cannot take logarithm of a negative number",
+	"input value must be",
+}
+
+var ignoredRegex *regexp.Regexp
```

Comment:
this can just be `var ignoredRegex = regexp.MustCompile(strings.Join(ignoredErrorPatterns, "|"))`